pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -72,6 +75,24 @@ protected void triggerShutdown() {
     // this will just notify all waiters that the scheduler is shutting down
     _monitor.enter();
     _monitor.leave();
+    _workerPool.shutdown();
+    try {
```

Comment:
nit: you can use Guava's `MoreExecutors#shutdownAndAwaitTermination` which handles this pattern pretty well

Reply:
Done